### PR TITLE
SLC Burst Product Baseline Stacking

### PR DESCRIPTION
**Pull request by SpicyGarlicAlbacoreRoll**

### CODE REVIEW

File: asf_search/ASFSearchOptions/validator_map.py [hunk 1]
```diff
@@ -59,4 +59,7 @@ def validate(key, value):
     'session':                parse_session,
     'host':                   parse_string,
     'provider':               parse_string,
+    'absoluteBurstID':        int,
+    'relativeBurstID':        int,
+    'fullBurstID':            parse_string
```
Comment (glshort) [inline asf_search/ASFSearchOptions/validator_map.py:64]: recommend `parse_string_list`, same as above

File: asf_search/search/baseline_search.py [hunk 2]
```diff
@@ -84,18 +84,22 @@ def get_stack_opts(
         raise ASFBaselineError(f'Requested reference product needs a baseline stack ID but does not have one: {reference.properties["fileID"]}')
 
     # build a stack from scratch if it's a non-precalc dataset with state vectors
-    if reference.properties['platform'] in [PLATFORM.SENTINEL1A, PLATFORM.SENTINEL1B]:
-        stack_opts.platform = [PLATFORM.SENTINEL1A, PLATFORM.SENTINEL1B]
+    if reference.properties['platform'].upper() in [PLATFORM.SENTINEL1A.upper(), PLATFORM.SENTINEL1B.upper()]:
```
Comment (glshort) [inline asf_search/search/baseline_search.py:87]: Good minor addition. Double-bagging: it is expected that the next revision of the dataset will have the normal style of platform name, capitalization-wise.

File: asf_search/search/baseline_search.py [hunk 3]
```diff
@@ -84,18 +84,22 @@ def get_stack_opts(
...
+        stack_opts.platform = [reference.properties['platform'], PLATFORM.SENTINEL1A, PLATFORM.SENTINEL1B]
```
Comment (glshort) [inline asf_search/search/baseline_search.py:88]: Isn't this just adding one of those two names that are already in the list by way of constants? I believe CMR will disregard capitalization differences, so any issues caused by that is purely at our end.

File: asf_search/search/search.py [hunk 4]
```diff
@@ -49,6 +49,9 @@ def search(
         relativeOrbit: Union[int, Tuple[int, int], Iterable[Union[int, Tuple[int, int]]]] = None,
         season: Tuple[int, int] = None,
         start: Union[datetime.datetime, str] = None,
+        absoluteBurstID: int = None,
+        relativeBurstID: int = None,
```
Comment (glshort) [inline asf_search/search/search.py:53]: Are these really ints? I thought there was more to the burst ID than just [0-9]. Would affect comments up above about `parse_int_list`, making them `parse_string_list`

File: asf_search/search/search_generator.py [hunk 5]
```diff
@@ -49,6 +49,9 @@ def search_generator(
         relativeOrbit: Union[int, Tuple[int, int], Iterable[Union[int, Tuple[int, int]]]] = None,
         season: Tuple[int, int] = None,
         start: Union[datetime.datetime, str] = None,
+        absoluteBurstID: int = None,
+        relativeBurstID: int = None,
+        fullBurstID: str = None,
```
Comment (glshort) [inline asf_search/search/search_generator.py:54]: Same comment as above re: burst IDs being strictly ints

File: tests/BaselineSearch/test_baseline_search.py [hunk 6]
```diff
@@ -13,7 +13,7 @@ def run_test_get_preprocessed_stack_params(product):
 
     original_properties = product['properties']
     
-    assert(params.processingLevel == [get_default_product_type(product['properties']['sceneName'])])
+    assert(params.processingLevel == [get_default_product_type(reference)])
```
Comment (glshort) [inline tests/BaselineSearch/test_baseline_search.py:16]: We'll revisit in the future if we want to even have a default for stacking or just always match the ref, as previously mentioned.

File: tests/BaselineSearch/test_baseline_search.py [hunk 7]
```diff
@@ -24,7 +24,7 @@ def run_test_get_unprocessed_stack_params(product):
 
     original_properties = product['properties']
     assert(original_properties['polarization'] in params.polarization)
-    assert(['VV', 'VV+VH'] == params.polarization)
+    assert(['VV', 'VV+VH'] == params.polarization if reference.properties['polarization'] in ['VV', 'VV+VH'] else ['HH','HH+HV'] == params.polarization)
```
Comment (glshort) [inline tests/BaselineSearch/test_baseline_search.py:27]: Be careful with bursts! Burst can't be dual-pol, so a VV+VH SLC will yield a bunch of VV bursts and a bunch of VH bursts, which can only be stacked with like-pol bursts. Will probably need some `if()` here to constrain it to the ref burst pol only if it's a burst stack.

**Review by glshort — CHANGES_REQUESTED** [review]
See comments in-line

File: asf_search/search/search.py (hunk 4 above)
Comment (SpicyGarlicAlbacoreRoll) [reply asf_search/search/search.py]: I'm pretty sure absolute and relative burst id are only ints (at least, I haven't run into any issues treating them as ints). Will have to ask and verify that this is the case.

Comment (SpicyGarlicAlbacoreRoll) [reply asf_search/search/search.py]: Just confirmed, relative and absolute burst IDs are integers

File: tests/BaselineSearch/test_baseline_search.py (hunk 7 above)
Comment (SpicyGarlicAlbacoreRoll) [reply tests/BaselineSearch/test_baseline_search.py]: Made it so burst product baseline params will only use the burst product's polarization

**Review by glshort — APPROVED** [review]

**Pull request merged**